client/src/pages/Opportunities/PostOpportunity/PostOpportunity.js

```diff
@@ -5,6 +5,7 @@ import Form from '../../../objects/Form/Form.js';
 import Alert from '../../..//components/Alert/Alert.js';
 import axios from 'axios';
 import parse from '../../../utils/parseFields';
+import valid from '../../../utils/validateForm';
 
 import endpoints from '../../../utils/endpoints.js';
```

Comment:
Here, we import /utils/endpoints.js. Endpoints.js contains data for our forms. It tells our <Form/> component how to render a list of <Input/> components.

Here is a visualization:
endpoints.js (data) --> <PostOpportunity --> <Form --> <Input

With this setup, a good place to keep our input tests is in **endpoints.js**

Here, you could define zipcodes like so:
{ name: 'zipcode', test: **_yourTestFunction() {...}_**, _otherField_... }

There are several ways to could implement tests. One way you could do it is like so:

1. User fills out info
2. User hits "submit"
3. **testAllFields() runs**
4. Passes ? Submit form else make all errors stand out with red and an X (bootstrap can do this).

You'll have to figure out
1. Who tells an Input to pass or not?
2. How to make PostOpportunity communicate correctly with your test function.

**I know that currently, our forms are kind of a mess. I've been trying to learn a good way to separate logic, views, and tests to make development DRY and readable. If you want, you could do another issue to get used to the codebase, then come back to this. Or we could get together and discuss the best way to handle forms.**